### PR TITLE
Feature/clan6 195

### PR DESCRIPTION
**Pull request by marcinkp**

### CODE REVIEW

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/CredentialsConfig.kt [hunk 1]
```diff
@@ -1,20 +1,6 @@
...
-    fun getCredentials(): JsonArray = load().getJsonArray(CREDENTIALS_ARRAY)
-
-    override fun validate(configJson: JsonObject): Boolean {
-        return CredentialsValidator.validate(configJson)
-    }
-
-    override fun filePath(): String = CREDENTIALS_CONFIG_FILE_PATH
-
-    override fun type(): ConfigType = ConfigType.CREDENTIALS
+class CredentialsConfig {
```
Comment (tMaxx) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/CredentialsConfig.kt]: `CredentialsConstants`? It doesn't configure anything now, just a bunch of constants ;)

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/EndpointsConfig.kt [hunk 2]
```diff
@@ -1,23 +1,8 @@
...
-    fun getEndpoints(): JsonArray = load().getJsonArray(ENDPOINTS_ARRAY)
-
-    override fun validate(configJson: JsonObject): Boolean {
-        return EndpointsValidator.validate(configJson)
-    }
-
-    override fun filePath(): String = ENDPOINTS_CONFIG_FILE_PATH
-
-    override fun type(): ConfigType = ConfigType.ENDPOINTS
+class EndpointsConfig {
```
Comment (tMaxx) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/EndpointsConfig.kt]: Same thing here, another config class that doesn't configure anything

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/controller/ConfirmationSender.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    private fun sendError(body: JsonObject) {
+        LOGGER.error("$ERROR_MESSAGE \nconfig:\n$body")
+        vertx.eventBus().send(CogboardConstants.EVENT_SEND_MESSAGE_TO_WEBSOCKET,
+                JsonObject().message(ERROR_MESSAGE))
+    }
+
+    private fun JsonObject.message(message: String): JsonObject {
+        return this
+                .put(CogboardConstants.PROP_EVENT_TYPE, PROP_EVENT_TYPE_NOTIFICATION_CONFIG_SAVE)
+                .put("message", message)
```
Comment (tMaxx) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/controller/ConfirmationSender.kt]: Consistency: you reference a lot of constants, but here you went with a string literal. I'd put it into a constant as well, it looks weird.

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/EndpointsService.kt [hunk 4]
```diff
@@ -1,24 +1,19 @@
...
 import com.cognifide.cogboard.config.utils.JsonUtils.putIfNotExist
 import com.cognifide.cogboard.config.validation.endpoints.EndpointsValidator
-import io.vertx.core.Vertx
 import io.vertx.core.json.JsonArray
 import io.vertx.core.json.JsonObject
 
-class EndpointsService(private val config: JsonObject, vertx: Vertx) {
-
-    private var storage: Storage = VolumeStorage(ConfigType.ENDPOINTS, vertx)
+class EndpointsService(private var storage: Storage, private val config: JsonObject = storage.loadConfig()) {
```
Comment (tMaxx) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/service/EndpointsService.kt]: Whitespace: in `WidgetRuntimeService` you add newlines for parameters, and here you don't. I'd add it here as well, just for clarity.

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/controller/ConfirmationSender.kt [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+import com.cognifide.cogboard.CogboardConstants
+import io.vertx.core.Vertx
+import io.vertx.core.json.JsonObject
+import io.vertx.core.logging.Logger
+import io.vertx.core.logging.LoggerFactory
+
+class ConfirmationSender(private val vertx: Vertx) {
+
+    fun confirmationAfter(action: (JsonObject) -> Boolean, body: JsonObject) =
```
Comment (szymon-owczarzak) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/controller/ConfirmationSender.kt]: Nice :)

Comment (szymon-owczarzak) [reply cogboard-app/src/main/kotlin/com/cognifide/cogboard/config/controller/ConfirmationSender.kt]: you should change order of attributes and make action the last one - then we can use `confirmationAfter(body) {
 action...
}

File: cogboard-app/src/main/kotlin/com/cognifide/cogboard/storage/VolumeStorage.kt [hunk 6]
```diff
@@ -1,40 +1,29 @@
...
-import com.cognifide.cogboard.config.Config
-import com.cognifide.cogboard.config.ConfigFactory
-import io.vertx.core.Vertx
 import io.vertx.core.json.JsonObject
 import io.vertx.core.logging.Logger
 import io.vertx.core.logging.LoggerFactory
 import java.io.File
 
-class VolumeStorage(configType: ConfigType, val vertx: Vertx) : Storage {
+class VolumeStorage(private val configType: ConfigType, private val path: String, private val validator: Validator) : Storage {
```
Comment (tMaxx) [inline cogboard-app/src/main/kotlin/com/cognifide/cogboard/storage/VolumeStorage.kt]: Newlines :)

**Pull request merged**